### PR TITLE
Bring msDrawQueryLayer() in line with msDrawVectorLayer() - addresses…

### PR DESCRIPTION
**Pull request by sdlime**
… #5744.

### CODE REVIEW

**Comment by sdlime** [comment]
@geographika, can you try testing again? I pushed the WMS filter execution below style handling so  that the right class is referenced within the query code. This should address your specific issue.

**Comment by geographika** [comment]
@sdlime - excellent! The style is now applied along with the filtering.

File: mapdraw.c [hunk 1]
```diff
@@ -1428,19 +1432,28 @@ int msDrawQueryLayer(mapObj *map, layerObj *layer, imageObj *image)
...
 
     if(cache) {
-      drawmode |= MS_DRAWMODE_SINGLESTYLE;
-      status = msDrawShape(map, layer, &shape, image, 0, drawmode); /* draw only the first style */
-    }
-    else
+      styleObj *pStyle = layer->class[shape.classindex]->styles[0];
+      if (pStyle->outlinewidth > 0) msOutlineRenderingPrepareStyle(pStyle, map, layer, image);
+      status = msDrawShape(map, layer, &shape, image, 0, drawmode|MS_DRAWMODE_SINGLESTYLE); /* draw only the first style */
+      if (pStyle->outlinewidth > 0) msOutlineRenderingRestoreStyle(pStyle, map, layer, image);
```
Comment (geographika) [inline mapdraw.c]: Is something meant to happen here?

File: mapwms.c [hunk 2]
```diff
@@ -1910,6 +1887,22 @@ this request. Check wms/ows_enable_request settings.",
...
+  if (filter) {
+    if (sld_url || sld_body) {
+      msSetError(MS_WMSERR,
+                 "Vendor-specific FILTER parameter cannot be used with SLD or SLD_BODY.",
+                 "msWMSLoadGetMapParams()");
+      return msWMSException(map, nVersion, NULL, wms_exception_format);
+    }
+
+    if (msWMSApplyFilter(map, nVersion, filter, need_axis_swap, wms_exception_format) == MS_FAILURE) {
+      return MS_FAILURE;/* msWMSException(map, nVersion, "InvalidFilterRequest"); */
```
Comment (geographika) [inline mapwms.c]: Is there a reason not to throw a msWMSException (I see this is from the original function).

File: mapdraw.c (hunk 1 above)
Comment (sdlime) [reply mapdraw.c]: I can't tell specifically what here means? The code follows the same pattern as msDrawVectorLayer() in terms of drawing when caching is required. We draw the first/bottom style (manipulating the style in cases where outline width is set. It looks right to me.

File: mapwms.c (hunk 2 above)
Comment (sdlime) [reply mapwms.c]: This is probably a better question for Dan. My solution was simply to move the query/filter code below style-handling block.

Comment (sdlime) [reply mapwms.c]: That's probably a question for Dan.

File: mapdraw.c (hunk 1 above)
Comment (geographika) [reply mapdraw.c]: Sorry - I was looking at it in the browser and it looked like an empty if statement without the brackets.

File: mapwms.c (hunk 2 above)
Comment (geographika) [reply mapwms.c]: If it is a simple MS_FAILURE then I thins servers just return a HTTP 500 error which are hard to debug compared to an error message? Maybe one for @dmorissette but not a blocker.

**Comment by sdlime** [comment]
@rouault, this is the first time I've tried using the backport function. I thought all that was needed was to set the appropriate label and then merge. Is that not true?

**Pull request merged**